tracks/javascript/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,77 @@
+This exercise explores string testing:
+- `RegExp`
+- `String` index accessor
+
+### Reasonable solutions
+
+This exercise can be approached with regular expressions, using the `String`
+prototype, or a combination of both. Below are consise solutions for either but
+combinations are valid too, as long as they make sense:
+
+- `isSilence`: Has no letters, digits or punctuation (ignoring whitespace).
+- `isShouting`: All letters are uppercase, and there is at least one letter.
+- `isAsking`: Ends with a question mark (ignoring whitespace).
+
+#### Regular expressions
+```javascript
+const isSilence = message => /^\s*$/.test(message)
+const isShouting = message => /^[^A-Za-z]*[A-Z]+(?:[^a-z]*)$/.test(message)
+const isAsking = message => /\?\s*$/.test(message)
+```
+
+#### String prototype
+```javascript
+const isSilence = message => message.trim() === ''
+const isShouting = message => message.toUpperCase() === message && message.toLowerCase() !== message
+const isAsking = message => message.trim()[message.length - 1] === '?'
+```
+
+#### Composition
+```javascript
+export const hey = (message) => {
+  if (isSilence(message)) {
+    return 'Fine. Be that way!'
+  }
+
+  if (isShouting(message)) {
+    return isAsking(message)
+      ? "Calm down, I know what I'm doing!"
+      : 'Whoa, chill out!'
+  }
+
+  return isAsking(message)
+    ? 'Sure.'
+    : 'Whatever.'
+}
+```
+
+Instead of ternaries, nested if statements are fine, as is a `switch` with each
```

Comment:
Do you mean nested ternaries? IMO ternary inside `if` is fine as long as it does not make the line too long.

Reply:
Sorry, I meant this as an alternative. In the composition it has "ternaries". Instead of that

```javascript
 if (isShouting(message)) {
    if (isAsking(message)) {
        return "Calm down, I know what I'm doing!"
    }
    return 'Whoa, chill out!'
}
``` 
This is fine too.

Reply:
Ok got it. I misread the statement.